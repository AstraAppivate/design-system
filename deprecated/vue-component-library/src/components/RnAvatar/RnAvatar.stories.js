import { storiesOf } from '@storybook/vue'

import RnAvatar from './index.vue'

storiesOf('Avatars', module)
  .add('Default', () => ({
    components: { RnAvatar },
    template: '<rn-avatar initials="RN" />',
  }))
  .add('Small', () => ({
    components: { RnAvatar },
    template: '<rn-avatar size="small" initials="RN" />',
  }))
  .add('Large', () => ({
    components: { RnAvatar },
    template: '<rn-avatar size="large" initials="RN" />',
  }))
