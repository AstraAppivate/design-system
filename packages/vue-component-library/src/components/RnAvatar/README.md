---
title: Avatar
description: Avatars are a great way to highlight to the user 
---

# Basic Example

```html
  <rn-avatar initials="RN" />
```

#### Props

Prop Name    | Value           | Required
------------ | --------------- | --------
**initials** | `{String}`      | false
**size**     | `small` `large` | false