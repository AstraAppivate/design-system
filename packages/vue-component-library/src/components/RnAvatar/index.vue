<template lang="html">
  <div class="rn-avatar" :class="size">
    <span class="initials">{{ initials }}</span>
  </div>
</template>
<script src="./scripts.js"></script>
