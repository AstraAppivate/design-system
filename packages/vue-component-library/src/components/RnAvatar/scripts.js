export default {
  name: 'RnAvatar',
  props: {
    initials: {
      type: String,
      required: true,
    },
    size: {
      type: String,
      default: 'default',
    },
  },
}
