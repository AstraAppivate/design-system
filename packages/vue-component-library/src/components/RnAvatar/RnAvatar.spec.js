import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import RnAvatar from '@/src/components/RnAvatar/index.vue'

describe('RnAvatar.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnAvatar, {
      propsData: {
        initials: 'zt',
      },
    })
  })

  it('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
