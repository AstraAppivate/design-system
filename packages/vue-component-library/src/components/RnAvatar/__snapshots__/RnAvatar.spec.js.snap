// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnAvatar.vue snapshot: has same HTML structure 1`] = `<div class="rn-avatar default"><span class="initials">zt</span></div>`;
